extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -101,7 +151,7 @@ private static QuarkusSecurityIdentity augmentIdentity(SecurityIdentity security
         // redirect_uri
         URI absoluteUri = URI.create(context.request().absoluteURI());
         String redirectPath = getRedirectPath(configContext, absoluteUri);
-        String redirectUriParam = buildRedirectUri(context, absoluteUri, redirectPath);
+        String redirectUriParam = buildUri(context, redirectPath);
```

Comment:
The refactoring from `buildRedirectUri` to `buildUri` itself is neat but we've just calculated `absoluteUri` above, URI calculation is quite expensive, so IMHO it would be more optimal to keep `buildRedirectUri` as is, and have `buildLogoutUri` as well which would also accept `absoluteUri` - a bit more verbose I agree :-), but would also be clearer IMHO.

Reply:
Yeah .. I missed changing that part to use the refactored method.

Maybe we should keep it cleaner now and improve accordingly if we have issues with performance?

But for this case in particular, I think we can use obtain the path directly from the `context.request()`, right ?

Reply:
But it was not really unclean - and obviously we won't be spending time measuring how `absoluteUri` calculation duplication affects the performance, but it is a duplication we can easily avoid

Reply:
Yeah. I've updated to just use `HttpServerRequest` methods. Like I previously mentioned, I guess it is OK, right ?

Reply:
Yeah it looks fine actually, sorry